test/test_native_mha.py

```diff
@@ -0,0 +1,288 @@
+import math
+
+import torch
+from torch.testing._internal.common_device_type import (
+    dtypes,
+    dtypesIfCUDA,
+    instantiate_device_type_tests,
+    onlyCUDA,
+    skipMeta,
+)
+from torch.testing._internal.common_utils import TestCase
+
+class TestMHADeviceType(TestCase):
+    @torch.inference_mode()
+    def _test_transform_bias_rescale_qkv_impl(
+        self, device, dtype, use_nt, use_padding=False
+    ):
+        tests = [
+            (64, 4, 16, 8),
+            # dim_per_head = 12 does not divide evenly by CPU vectorization length of 8
+            (24, 2, 4, 2),
+            # Make sure CUDA can handle small input sizes
+            (2, 2, 2, 2),
+            # dim_per_head = 6 does not divide evenly by CUDA vectorization length of 4, causes alignment issues
+            (24, 4, 4, 2),
+            (48, 4, 16, 8),
+        ]
+        for (embed_dim, num_heads, bs, sl) in tests:
```

Comment:
nit: you can modestly improve debuggability when a test fails by using subTest to report which parameters were chosen for this test